keras_core/models/functional.py

```diff
@@ -154,13 +154,12 @@ def __init__(self, inputs, outputs, name=None, **kwargs):
             self.trainable = trainable
 
         self._layers = self.layers
-        self.built = True
+        self.build(None)
```

Comment:
Not totally sure about this. I think it would be a nice property to get our models and layers all hitting `build()` at some point, so we don't have to keep "post build actions" in sync across classes.

This will now lock the functional state after construction, similar to calling build on a layer. But do we actually want that? Safer to stick to original code here? (I think both work)